app/src/androidTest/java/com/github/sdpcoachme/messaging/InAppNotifierTest.kt

```diff
@@ -145,6 +148,75 @@ class InAppNotifierTest {
         }
     }
 
+    @Test
+    fun onMessageReceivedEnablesEmailRecoveryAndOpensChatActivityWhenUserReceivesNotificationThenLoggsOutAndThenClicksOnIt() {
+        val intent = Intent(ApplicationProvider.getApplicationContext(), MapActivity::class.java)
+
+        // The start screen can be anything, MapActivity is just used here
+        ActivityScenario.launch<MapActivity>(intent).use {
+
+            sendNotification("Title", "Body", personalChatId, "messaging")
+
+
+            composeTestRule.onNodeWithTag(Dashboard.TestTags.DRAWER_HEADER, useUnmergedTree = true)
+                .assertIsNotDisplayed()
+            composeTestRule.onNodeWithTag(Dashboard.TestTags.Buttons.HAMBURGER_MENU, useUnmergedTree = true)
+                .performClick()
+            composeTestRule.onNodeWithTag(Dashboard.TestTags.DRAWER_HEADER, useUnmergedTree = true)
+                .assertIsDisplayed()
+
+            val device = UiDevice.getInstance(getInstrumentation())
+            device.waitForIdle()
+            device.findObject(
+                UiSelector().text("Log out"))
+                .click()
+
+            // exit application
+            device.pressHome()
+
+            clickOnNotification("Title", "Body")
+
+            // Check if ChatActivity is opened
+            composeTestRule.onNodeWithText(toUser.firstName + " " + toUser.lastName).assertExists()
+            composeTestRule.onNodeWithTag(CONTACT_FIELD.LABEL, useUnmergedTree = true).assertExists()
+            composeTestRule.onNodeWithTag(CHAT_FIELD.LABEL, useUnmergedTree = true).assertExists()
+        }
+    }
+
+    @Test
+    fun onMessageReceivedEnablesEmailRecoveryAndOpensContactsListWhenUserReceivesNotificationWithoutChatIdThenLoggsOutAndThenClicksOnIt() {
```

Comment:
Good that you thought about this case too. I think, once the above test has a well-explanatory comment, this test will be understandable too.